lib/seahorn/HornCex.cc

```diff
@@ -220,11 +227,25 @@ namespace seahorn
               << ". Expected .xml, .ll, or .bc.\n";
     }
 
-    if (!SliceOutputFile.empty()) {
-      sliceErrorTrace(F, trace);
-      dumpLLVMBitcode(M, SliceOutputFile.c_str());
+    if (!BmcSliceOutputFile.empty()) {
+      llvm::DenseSet<const llvm::BasicBlock*> region(trace.getRegion());
+      reduceToRegion(F, region, false);
+      dumpLLVMBitcode(M, BmcSliceOutputFile.c_str());
       return true;
     }
+
+    if (!CpSliceOutputFile.empty())
+    {
+      DenseSet<const BasicBlock*> region;
+      for (const CutPoint* cp:cpTrace)
+      for (CutPoint::const_iterator it=cp->succ_begin();it!=cp->succ_end();it++)
+      for (CpEdge::iterator ite=(*it)->begin();ite!=(*it)->end();ite++)
+        region.insert(&*ite);
+      reduceToRegion(F,region,false);
```

Comment:
I looked at the original `reduceToRegion` again. I don't see how it can fail because of `PHI` nodes. Only if the region argument to it is not really a region. Using `DeleteDeadBlock` is not right since we might be deleting regions with loops. Not deleting basic blocks probably only masks the actual problem and if we won't fix it here, it will be harder to debug later on.

I do not understand your comment about *abstraction*. There is no abstraction happening here since we are working on the exact code being analyzed. That code might be an abstraction of something else, but that does not matter. 

Please send me an example that fails in `reduceToRegion` when basic blocks are removed. Maybe I can come up with an alternative explanation.